app/javascript/packs/articlePage.jsx

```diff
@@ -71,6 +72,34 @@ function showAnnouncer() {
   document.getElementById('article-copy-link-announcer').hidden = false;
 }
 
+// Temporary Ahoy Stats for comment section clicks on controls
+function trackCommentsSectionClicks() {
+  document.querySelectorAll('[data-tracking-name]').forEach((entry) => {
+    entry.addEventListener('click', (_event) => {
+      ahoy.track('Comment section click', {
+        page: location.href,
+        element: entry.getAttribute('data-tracking-name'),
+      });
+    });
+  });
+}
```

Comment:
Given some posts will have a lot of comments, I think it would be an enhancement to only add one event listener, e.g.:

```suggestion
function trackCommentsSectionClicks() {
  document
    .getElementById('comments')
    .addEventListener('click', ({ target }) => {
      // We check for any parent container with a data-tracking-name attribute, as otherwise
      // SVGs inside buttons can cause events to be missed
      const relevantNode = target.closest('[data-tracking-name]');

      if (!relevantNode) {
        // We don't want to track this click
        return;
      }

      ahoy.track('Comment section click', {
        page: location.href,
        element: relevantNode.dataset.trackingName,
      });
    });
}
```

I've added `target.closest('[data-tracking-name]')` because the `target` is wherever the click event _started_ which is sometimes an icon or whatever inside the button, rather than the element with the data attribute we're looking for 🙃

Reply:
`target.closest`?! Very cool! 🤯 (Can you tell I'm a JavaScript newbie?). I see now how this sidesteps the "new elements are added to the page at runtime without a handler" issue: `comments` **will** be on the page at runtime/load!